### ISSUE TITLE
 Failure to build with --prod when importing  json file in environment.prod file

### ISSUE THREAD

**Issue by cvgaviao** [angular/angular-cli#9203]
### Versions
``` 
Angular CLI: 1.6.4
Node: 8.9.1
OS: linux x64
Angular: 5.2.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker

@angular/cdk: 5.0.4
@angular/cli: 1.6.4
@angular/flex-layout: 2.0.0-beta.12
@angular/material: 5.0.4
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.48
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.4
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.5.3
webpack: 3.10.0
```

### Repro steps
* Step 1
Modify the typings.d.ts and include the following to allow json import:
``` 
declare module "*.json" {
  const value: any;
  export default value;
}
```
* Step 2
Modify the environment.prod :
```
import env from './env.json';
...
export const environment = {
  production: true,
  version: env.npm_package_version,
```
* Step 3
build the project with:
` ng build --prod`

### Observed behavior
```
Cannot read property 'npm_package_version' of undefined
TypeError: Cannot read property 'npm_package_version' of undefined
    at Object.p5Ee (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:182643)
    at t (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:220)
    at Object.aR8+ (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:128006)
    at t (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:220)
    at Object.JwoV (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:97570)
    at t (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:220)
    at Object.Zq8w (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:118840)
    at t (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:220)
    at Object.0 (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:2685)
    at t (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:220)
    at e.+Mol.Object.defineProperty.value (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:568)
    at Object.<anonymous> (/home/projects/c8tech-pwa/dist-server/main.1dc033d7e120ce685531.bundle.js:1:577)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
error Command failed with exit code 1.

```

### Desired behavior
The build for production to succeed as it does with ng build

### Mention any other details that might be useful (optional)
The env.json file is being generated by env2json before starting the build and has the format bellow:
`{
  "npm_package_version": "0.1.0"
}`

**Comment by ArthurianX** [comment]
@cvgaviao How did you solve this, for `--prod` build. I'm stuck here.

**Comment by filipesilva** [comment]
Thanks for reporting this issue. This issue was originally reported a long time ago and since there we've had many releases, one of which might have addressed this problem. Please update to the most recent Angular CLI version.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.
